database/src/monero.rs

```diff
@@ -0,0 +1,198 @@
+//! Abstracted Monero database operations; `trait Monero`.
+
+//---------------------------------------------------------------------------------------------------- Import
+#[allow(unused_imports)] // FIXME: these traits will be eventually in the function impls.
+use crate::{
+    database::Database,
+    env::Env,
+    table::Table,
+    transaction::{RoTx, RwTx},
+    ConcreteEnv,
+};
+
+//---------------------------------------------------------------------------------------------------- Monero
+/// Monero database operations.
+///
+/// This trait builds on top of:
+/// - [`Env`]
+/// - [`Table`]
+/// - [`Database`]
+/// - [`RoTx`], [`RwTx`]
+///
+/// to provide higher-level `Monero`-specific database operations.
+///
+/// # Notes
+/// Things to note about this trait:
+/// 1. All methods are already implemented
+/// 2. [`ConcreteEnv`] automatically implements [`Monero`]
+///
+/// This means that these methods can be
+/// called directly on `ConcreteEnv`.
+///
+/// Normal `tower::Service` readers/writers will still go
+/// through messages but the underlying database thread will
+/// use these functions.
+///
+/// Underlying database backends can re-implement any of
+/// these functions if the generic version is not good enough.
+///
+/// # TODO
+/// TODO: What is a good name for this trait?
+///
+/// TODO: These functions should pretty much map 1-1 to the `Request` enum.
+///
+/// TODO: These are function names from `old_database/` for now.
+/// The actual underlying functions (e.g `get()`) aren't implemented.
+#[allow(missing_docs)]
+pub trait Monero: Env {
```

Comment:
I don't think this trait is needed, I think these methods should just be functions which take in a generic `RoTx`/`RwTx`. This would mean an underlying DB can't re-implement a method in a more optimized way for that back-end, but I don't think this is an issue as if our `Table` trait contain enough info (more on that in another comment) then the back-end can optimize for generic tables with certain characteristics.